Optimization / Multipoint changes

tommyka1 opened the pull request:
**-** Replaced current optimizations (multipoint / backtrack) with new "Low Performance Mode" (much more dynamic and less impactful to ragebot performance):

- Low performance mode will adjust the amount of backtrack ticks processed based on your fps (maximum value achieved when fps is greater than or equal to 2 * (the server's tickrate), minimum value of 0 achieved when fps is less than or equal to the server's tickrate).
- Low performance mode will also adjust the amount of hitboxes (exluding the head hitbox) that are processed by the multipoint function (maximum value of 15 achieved when fps is greater than or equal to 2 * (the server's tickrate), minimum value of 0 achieved when fps is less than or equal to the server's tickrate). 
- In critical cases, low performance mode will prevent limb scanning.

**-** Added feet as a selectable ragebot hitbox group.
**-** Added hands to the arms hitbox group.
**-** Scaled multipoint into two options: head multipoint and body multipoint.

Commit:
Optimization / Multipoint changes
- Replaced current optimizations (multipoint / backtrack) with new "Low Performance Mode" (much more dynamic and less impactful to ragebot performance):
    - Low performance mode will adjust the amount of backtrack ticks processed based on your fps (maximum value achieved when fps is greater than or equal to 2 * (the server's tickrate), minimum value of 0 achieved when fps is less than or equal to the server's tickrate).
    - Low performance mode will also adjust the amount of hitboxes (exluding the head hitbox) that are processed by the multipoint function (maximum value of 15 achieved when fps is greater than or equal to 2 * (the server's tickrate), minimum value of 0 achieved when fps is less than or equal to the server's tickrate).
    - In critical cases, low performance mode will prevent limb scanning.
- Added feet as a selectable ragebot hitbox group.
- Added hands to the arms hitbox group.
- Scaled multipoint into two options: head multipoint and body multipoint.

Commit:
Optimization / Multipoint changes # 2
- Scaled low performance mode into 2 options (ragebot / backtrack) to allow the use of it while legit cheating

tommyka1 commented:
**-** Scaled low performance mode into 2 options (ragebot / backtrack) to allow the use of it while legit cheating.

notgoodusename commented:
I like your idea about not processing certain body parts depending off FPS
The scaling multipoint seems to work, since low FPS disables ragebot altogether, but i prefer to have ragebot work 24/7 even in low fps situations
Scalling backtrack is a bad idea, since it would remove ticks, which the ragebot doesnt use every single one, it only takes the last possible tick
Also the neck not being scanned is on purpose

tommyka1 commented:
Scaling backtrack does remove ticks, but it's much better than backtrack being ignored alltogether when fps is low. Also, ragebot should still work properly in low fps situations, it will just ignore limbs (head and body will still be targeted). If you think that's too much though, i can delete that part and keep the scaling multipoint, which gives a pretty nice fps boost when several enemies are scanned. (Also, i'm aware that neck isn't being scanned on purpose)

notgoodusename commented:
> Scaling backtrack does remove ticks, but it's much better than backtrack being ignored alltogether when fps is low. 

Its better to disable backtrack because with backtrack enabled you are running ragebot twice on every player, scalling ticks down doesnt stop this, it just makes the rage backtrack worse

>Also, ragebot should still work properly in low fps situations, it will just ignore limbs (head and body will still be targeted). If you think that's too much though, i can delete that part and keep the scaling multipoint, which gives a pretty nice fps boost when several enemies are scanned. (Also, i'm aware that neck isn't being scanned on purpose)

https://streamable.com/8yw86l

tommyka1 commented:
This shouldn't be happening, I'll push a new version addressing that bug and reverting the backtrack changes soon.

Commit:
Optimization / Multipoint changes #3
- Restored old low performance backtrack function.
- Fixed hitscan bugs when using low performance mode.

tommyka1 commented:
Everything should be working fine now. I've restored the backtrack optimizations to their original version, and fixed some other issues.
The ragebot will always target the head and body hitboxes, and will also target the limbs if they're the only selected hitboxes or the user's fps is higher than the server's tickrate.
The multipoint will always scan every head point, and will also scan the points of other body parts depending on how much fps the user has (multipoint will work on every hitbox if the user has fps equal to or greater than 2 * (the server's tickrate), and will only work on the head hitbox if the user's fps is less or equal to the server's tickrate. Between those values, it will dynamically scan a certain amount of body parts, going from top to bottom).

Commit:
Optimization / Multipoint changes #4
- small hotfix.

Liam5750 commented:
I've found threading to work great for me too but I love the idea of not disabling non essential Ragebot features but just kinda turning them down in a sense. Great pr

Liam5750 commented:
My fps after adding this went in the negatives. Great pr!
